program/mobilenets-armcl-opencl/mobilenet.cpp

```diff
@@ -126,9 +137,12 @@ void setup_mobilenet(GraphObject& graph,
                      float multiplier,
                      const std::string& weights_dir,
                      const float *input_data_buffer,
-                     float *output_data_buffer)
+                     float *output_data_buffer,
+                     CKDataLayout data_layout)
 {
-    TensorShape input_shape(image_size, image_size, 3U, 1U);
+    TensorShape input_shape = (data_layout == LAYOUT_NCHW) ?
```

Comment:
Doesn't this contradict the above comment that:
```
// Source data layout is always NCHW
```

Reply:
It doesn't. Source data mentioned above is an image we load and it is always `NCHW`. But processing data can be `NHWC` and input tensor should have such dimensions order. And we optionally convert `NCHW -> NHWC` in input data accessor.